### ISSUE TITLE
[ASSETS] MigrateFileTask gives error "Extension is not allowed" when migrating files from 3.x to 4.x

### ISSUE THREAD

**Issue by hdpero** [silverstripe/silverstripe-framework#8312]
## Affected Version

- SilverStripe 4+ _(probably all of the releases)_ , tested on 4.1.2 and 4.2.1
- **OS:** Windows 10, running PHP 7.1.9 on Apache 2.4.27 and MySQL 5.7.19 

## Description

When I started a MigrateFileTask from old SS3 project assets to new SS4 assets I got an error saying `"Extension is not allowed"`, like shown in screenshot.

![error-migrate-file](https://user-images.githubusercontent.com/13778690/43934412-88cb6bd8-9c4f-11e8-8ced-623d4e8b8516.png)

I've tried to add an extension which were caused an error to appear under **app/_config/config.yml** 
```yml
SilverStripe\Assets\File:
   allowed_extensions:
     - pdf
     - zip
```
And I tried with variations: "- pdf" , "- .pdf", "- .pdf" but nothing seemed to work, and even under `SilverStripe\Assets\Image` but it didn't work. 

I've tried running it both in CLI and browser, same error appears. 

## Temporary solution
What I temporary did was I deleted couple of .pdf and .zip that was part of the assets in old SS3 _(all other file types where images - .jpg, .jpeg, .gif and .png)_, and run a MigrateFileTask again which has completed successfully without a problem.

## Steps to Reproduce

- Install a fresh SilverStripe 3 release (latest 3.7.1 for example) 
- Upload some files in SS3 version (include files like .pdf , .doc , .zip , not only Image file types)
- Install a fresh SilverStripe 4 receipe
- Run a MigrateFileTask to upgrade Files to new schema
- It gives a error as written above

## PR

- [ ] https://github.com/silverstripe/silverstripe-assets/pull/170

**Comment by bigSiebs** [comment]
I encountered this same issue when upgrading from an older SS3 site (3.1). In my case, I had three PDF files that had the ClassName `SilverStripe/Assets/Image` in their File records. I changed it to `SilverStripe/Assets/File`, and that seemed to resolve the issue.

**Comment by maxime-rainville** [comment]
Fixed by silverstripe/silverstripe-assets#170
